### ISSUE TITLE
Give access to root object in serializer

### ISSUE THREAD

**Issue by P3ridot** [OkaeriPoland/okaeri-configs#38]
I'd like to access root object in serializer. I'll give an exmaple:

I have object... For eg. ComplexMessage that contains ChatMessage & ActionbarMessage . In normal serializer I would serialize it to something like that:
```yaml
message:
  chat: "Chat message"
  actionbar: "Actionbar message"
```
which is fine when I have 2 messages types used

But when I have only 1 message type (for eg. chat) I ends up with 
```yaml
message:
  chat: "Chat message"
```

but I wants it's to be simpler - and serialize it as String:
```yaml
message: "Chat message"
```

**Comment by PinkLolicorn** [comment]
Try `4.0.10-beta.1`.

```java

**Comment by P3ridot** [comment]
> Note that getAsList and getAsMap analogs are currently unavailable.
So currently I can achieve 
```java
message: "String"
```
but not:
```java
message: 
- "Line 1"
- "Line 2"
```
Right?

**Comment by PinkLolicorn** [comment]
The `getAsList` and the `getAsMap` is only a shortcut to a specific `Configurer#resolveType` call. These can always be performed using `getDirect(key, GenericsDeclaration)` or `getValueDirect(GenericsDeclaration)`.

The `getValueAsMap` method will never make sense because if the object is interpretable as Map it is deconstructed and passed as standard DeserializationData. The `getValueAsList` map will be available in the release version.

```java
    public <T> T getDirect(@NonNull String key, @NonNull GenericsDeclaration genericType) {
        Object object = this.data.get(key);
        return (T) this.configurer.resolveType(object, null, genericType.getType(), genericType, SerdesContext.of(this.configurer));
    }
```
```java
    public <T> T getValueDirect(@NonNull GenericsDeclaration genericType) {
        return this.getDirect(ObjectSerializer.VALUE, genericType);
    }
```

There is nothing preventing you from calling `getAsList(ObjectSerializer.VALUE, String.class)` which will be the implementation of `getValueAsList(Class)`.

```java
    public <T> List<T> getAsList(@NonNull String key, @NonNull Class<T> listValueType) {
        GenericsDeclaration genericType = GenericsDeclaration.of(List.class, Collections.singletonList(listValueType));
        return (List<T>) this.getAsCollection(key, genericType);
    }
```
```java
    public <T> Collection<T> getAsCollection(@NonNull String key, @NonNull GenericsDeclaration genericType) {
        if (!Collection.class.isAssignableFrom(genericType.getType())) {
            throw new IllegalArgumentException("genericType.type must be a superclass of Collection: " + genericType);
        }
        return this.getDirect(key, genericType);
    }
```

**Comment by PinkLolicorn** [comment]
> The `getValueAsList` will be available in the release version.

Well, the `getValueAsList(Class)` is now available in `4.0.10-beta.2` since I decided `Deserialization#hasValue` should be renamed to `Deserialization#isValue` to avoid potential confusion.

**Comment by PinkLolicorn** [comment]
Mostly complete API available staring from `5.0.0-beta.1`.

![image](https://user-images.githubusercontent.com/44551064/211689688-b3a138de-c100-4cbe-9271-dcf1243b84f3.png)

![image](https://user-images.githubusercontent.com/44551064/211689783-1a426e2d-37be-426e-8b2f-cad48696eb34.png)
